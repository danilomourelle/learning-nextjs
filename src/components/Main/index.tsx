const Main = () => {
  return (
    <main className="main">
      <h1>React AvanÃ§ado</h1>
    </main>
  );
};

export default Main;
