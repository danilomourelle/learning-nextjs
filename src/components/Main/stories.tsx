import { Meta, StoryObj } from "@storybook/react";
import Main from ".";

const meta: Meta<typeof Main> = {
  title: "Main",
  component: Main,
};

export default meta;

export const Default: StoryObj = {};
