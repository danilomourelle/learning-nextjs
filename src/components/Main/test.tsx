import { render, screen } from "@testing-library/react";

import Main from ".";

describe("<Main />", () => {
  it("should render the head", () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole("heading", { name: "React AvanÃ§ado" }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render the colors", () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toHaveStyle({ "background-color": "#06092b" });
  });
});
