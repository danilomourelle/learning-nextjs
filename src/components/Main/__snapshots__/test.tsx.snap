// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the head 1`] = `
<main
  class="sc-f59cca65-0 dsmJwL"
>
  <h1>
    React Avan√ßado
  </h1>
</main>
`;
